src/main/java/ai/Storage.java

```diff
@@ -0,0 +1,103 @@
+package ai;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import ai.exception.AiException;
+import ai.task.Deadline;
+import ai.task.Event;
+import ai.task.Task;
+import ai.task.ToDo;
+
+/**
+ * Deals with file writing and reading.
+ */
+public class Storage {
+    private ArrayList<Task> tasks = new ArrayList<Task>();
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the file using the given filePath and scans for each line.
+     *
+     * @return ArrayList of type Task.
+     * @throws IOException if .txt file cannot be created successfully.
+     * @throws AiException if the line in the .txt file cannot be read.
+     */
+    public ArrayList<Task> load() throws IOException, AiException {
+        Path path = Paths.get(filePath);
+        Path directory = path.getParent();
+
+        File dir = new File(directory.toString());
+
+        dir.mkdirs();
+
+        if (!dir.exists()) {
+            System.out.println("Path created");
+        }
+
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+
+
+        f.createNewFile();
+        // Check if file exists
+        if (!f.exists()) {
+            // Create file
+            System.out.println("File created");
+        }
+
+        // File exists, read from file
+        while (s.hasNext()) {
+            tasks.add(readFileLine(s.nextLine()));
+        }
+
+        s.close();
+        return tasks;
+    }
+
+    /**
+     * Reads the given input, parses it into the correct format to initialise the Task to be returned.
+     *
+     * @param input String of a single line in the .txt file.
+     * @return Task in the correct type.
+     * @throws AiException if the lines cannot be read.
+     */
+    public Task readFileLine(String input) throws AiException {
```

Comment:
Good that you extracted the process of reading file lines into a separate method!